workflows/wf_theiacov_illumina_pe.wdl

```diff
@@ -38,92 +45,130 @@ workflow theiacov_illumina_pe {
       adapters = adapters,
       phix = phix
   }
-  call align.bwa {
-    input:
-      samplename = samplename,
-      read1 = read_QC_trim.read1_clean,
-      read2 = read_QC_trim.read2_clean,
-      reference_genome = reference_genome
-  }
-  if (trim_primers){
-    call consensus_call.primer_trim {
+  if (organism != "flu"){
+    call align.bwa {
       input:
         samplename = samplename,
-        primer_bed = select_first([primer_bed]),
-        bamfile = bwa.sorted_bam
+        read1 = read_QC_trim.read1_clean,
+        read2 = read_QC_trim.read2_clean,
+        reference_genome = reference_genome
     }
-    call assembly_metrics.stats_n_coverage as stats_n_coverage_primtrim {
-    input:
-      samplename = samplename,
-      bamfile = primer_trim.trim_sorted_bam,
-    }
-  }
-  call consensus_call.variant_call {
-    input:
-      samplename = samplename,
-      bamfile = select_first([primer_trim.trim_sorted_bam,bwa.sorted_bam]),
-      reference_genome = reference_genome,
-      variant_min_depth = min_depth
-  }
-  call consensus_call.consensus {
-    input:
-      samplename = samplename,
-      bamfile = select_first([primer_trim.trim_sorted_bam,bwa.sorted_bam]),
-      reference_genome = reference_genome,
-      consensus_min_depth = min_depth
-  }
-  call consensus_qc_task.consensus_qc {
-    input:
-      assembly_fasta = consensus.consensus_seq,
-      reference_genome = reference_genome
-  }
-  call assembly_metrics.stats_n_coverage {
-    input:
-      samplename = samplename,
-      bamfile = bwa.sorted_bam
-  }
-  if (organism == "sars-cov-2") {
-    # sars-cov-2 specific tasks
-    call taxon_ID.pangolin4 {
+    if (trim_primers){
+      call consensus_call.primer_trim {
+        input:
+          samplename = samplename,
+          primer_bed = select_first([primer_bed]),
```

Comment:
Since `primer_bed` is an optional output, we're technically feeding a `File?` to the `samplename` `File` input which WDL won't allow (i.e. it can't coerce `File` to `File?`). Using `select_first()` is a cheeky work around to enable use of `File?` for `File` inputs. Worth noting that this should only be used with an understanding of the workflow logic as it can cause failures if that output is never called. But since we nest it in the same conditional that will create `primer_bed`, we won't run into that issue.